client/src/renderer/components/Clipboard/index.jsx

```diff
@@ -0,0 +1,104 @@
+/** @format */
+
+import "./styles.scss";
+import { Component, h } from "preact";
+import clipboard from "electron-clipboard-extended";
+
+/**
+ * Stores and lists contents of system clipboard
+ * @extends Component
+ */
+
+export default class Clipboard extends Component {
+  /**
+   * Constructs a new Clipboard component
+   * @param {Object} props Component properties
+   */
+  constructor(props) {
+    super(props);
+    this.state = { clipboardStack: [] };
+    this.handleTextChanged = this.handleTextChanged.bind(this);
+    this.handleImageChanged = this.handleImageChanged.bind(this);
+  }
+
+  componentDidMount() {
+    clipboard.startWatching();
+    clipboard.on("text-changed", this.handleTextChanged);
+    clipboard.on("image-changed", this.handleImageChanged);
+  }
+
+  componentWillUnmount() {
+    clipboard.stopWatching();
+    clipboard.off("text-changed");
+    clipboard.off("image-changed");v
+  }
+
+  /**
+   * Reacts to changes in system clipboard
+   * @listens {text-changed} listens for text change event
+   */
+  handleTextChanged() {
+    this.setState(prevState => {
+      return {
+        clipboardStack: [
+          ...prevState.clipboardStack,
+          new ClipboardItem("txt", clipboard.readText()),
+        ],
+      };
+    });
+  }
+  /**
+   * Reacts to changes in system clipboard
+   * @listens {text-changed} listens for text change event
+   */
+  handleImageChanged() {
+    this.setState(prevState => {
+      const img = clipboard.readImage();
+      const imgURI = img.toDataURL();
+      return {
+        clipboardStack: [
+          ...prevState.clipboardStack,
+          new ClipboardItem("img", imgURI),
+        ],
+      };
+    });
+  }
+
+  /**
+   * Renders component
+   * 
+   * Returns unordered list, each list item corresponding to some clipboard datum and the button to remove the item
+   * @example
+   * <ul>
+   * ...
+   *   <li>
+   *   <button>{remove item from list}</button>
+   *   <p>{clipboard text}</p>
+   *   </li>
+   * ...
+   * </ul>
+   */
+  render(props, state) {
+    return (
+      <ul class="clipboard">
+        {state.clipboardStack.map((item, index) => (
+          <li class="clipboard__item" id={"cp_it_" + index}>
```

Comment:
Do not use `id` here. It's *super* rare that you should ever actually use ids at all. For CSS, always prefer classes because they have lower specificity. And for encoding data like you have above, it would be better to stick the index in an attribute. Just make sure to prefix it with `data-` like so:

```suggestion
          <li class="clipboard__item" data-index={index}>
```